### PR TITLE
Add support to SnappableHandler to pick from other layers

### PR DESCRIPTION
**Pull request by austintang**
I introduced one new modeConfig property:
* `layerIdsToSnapTo: ?(string[])`

If `layerIdsToSnapTo` is present, the `SnappableHandler` will look for picks in the specified layers in addition to features in the current `SnappableHandler._handler`. If `layerIdsToSnapTo` isn't defined, the handler will default to picking from the current `SnappableHandler._handler`.

To see this feature in action, paste the following code snippet in the `render()` method (right before the method returns) in the file `examples/deck/example.js`:
```js
    layers.push(
      new EditableGeoJsonLayer({
        id: 'test-layer',
        selectedFeatureIndexes: [],
        fp64: false,
        getFillColor: [255, 0, 0, 255],
        data: [
          {"type":"Feature","properties":{},"geometry":{"type":"Polygon","coordinates":[[[-122.3952996456623,37.877279091381524],[-122.34470916032794,37.877279091381524],[-122.34470916032794,37.85118811159345],[-122.3952996456623,37.85118811159345],[-122.3952996456623,37.877279091381524]]]}}
        ]
      })
    );

    layers.push(
      new EditableGeoJsonLayer({
        id: 'test-layer2',
        selectedFeatureIndexes: [],
        fp64: false,
        getFillColor: [0, 255, 0, 255],
        data: [{"type":"Feature","properties":{},"geometry":{"type":"Polygon","coordinates":[[[-122.51409022077985,37.869217283494564],[-122.55434977639163,37.88652710247724],[-122.56327414956525,37.86102193657733],[-122.54391170517715,37.854042309973],[-122.54390179262947,37.81944995034826],[-122.51973008097127,37.84256451196541],[-122.4808255716967,37.83945728896286],[-122.47966972189454,37.861229451757495],[-122.4797338087995,37.88959953265049],[-122.51969204410665,37.889800617306385],[-122.52789951696104,37.90947991368635],[-122.54375610750321,37.89454288093134],[-122.51409022077985,37.869217283494564]]]}}]
      })
    );
```

and set the `EditableGeoJsonLayer` constructor modeConfig to:
```js
{
  ...modeConfig,
  layerIdsToSnapTo: ['test-layer', 'test-layer2']
}
```

Note that the green and red polygons live in their own separate layers.

### CODE REVIEW

File: modules/layers/src/mode-handlers/snappable-handler.js [hunk 1]
```diff
@@ -103,12 +103,42 @@ export class SnappableHandler extends ModeHandler {
     }
   }
 
+  // If pickFromOtherLayerIds is present in modeConfig, this method will return the features
+  // from the specified layers. Additionally, if appendPicksFromOtherLayers is set to true
+  // in modeConfig, the features from other layers will be appended to the features in
+  // this._handler. Otherwise, this method will simply return the features from this._handler.
+  _getFeaturesFromRelevantLayer(): any[] {
```
Comment (georgios-uber) [inline modules/layers/src/mode-handlers/snappable-handler.js]:
```suggestion
  _getFeaturesFromRelevantLayer(): Object[] {
```

File: modules/layers/src/mode-handlers/snappable-handler.js [hunk 2]
```diff
@@ -103,12 +103,42 @@ export class SnappableHandler extends ModeHandler {
...
+    if (pickFromOtherLayerIds) {
+      const otherLayersToPickFrom = this._context.layerManager.layers.filter(
+        layer => pickFromOtherLayerIds && pickFromOtherLayerIds.includes(layer.id)
+      );
+
+      features = otherLayersToPickFrom
+        .map(otherLayer => otherLayer.props.data)
+        .reduce((a, b) => [...a, ...b], []);
+
+      if (!appendPicksFromOtherLayers) {
```
Comment (georgios-uber) [inline modules/layers/src/mode-handlers/snappable-handler.js]: Why do you need to check both?

File: modules/layers/src/mode-handlers/snappable-handler.js [hunk 3]
```diff
@@ -103,12 +103,42 @@ export class SnappableHandler extends ModeHandler {
...
+
   _getNonPickedIntermediateHandles(): EditHandle[] {
     const handles = [];
-    const { features } = this._handler.featureCollection.getObject();
+    const features = this._getFeaturesFromRelevantLayer();
+    const { pickFromOtherLayerIds, appendPicksFromOtherLayers } = this._modeConfig || {};
 
     for (let i = 0; i < features.length; i++) {
-      if (!this._handler.getSelectedFeatureIndexes().includes(i) && i < features.length) {
+      const isIndexSelected =
```
Comment (georgios-uber) [inline modules/layers/src/mode-handlers/snappable-handler.js]: This is a bit hard to read...

**Comment by georgios-uber** [comment]
Would be nice to have some docs :)

File: modules/layers/src/mode-handlers/snappable-handler.js (hunk 2 above)
Comment (austintang) [reply modules/layers/src/mode-handlers/snappable-handler.js]: Are you saying that we don't need the `appendPicksFromOtherLayers` modeConfig property, such that if `pickFromOtherLayerIds` property is populated, automatically pick from those layers in addition to the current layer?

**Pull request merged**